src/D2L.Security.OAuth2/OAuth2Factory.cs

```diff
@@ -0,0 +1,38 @@
+﻿using D2L.Security.OAuth2.Keys;
+using D2L.Security.OAuth2.Keys.Default;
+using D2L.Security.OAuth2.Utilities;
+
+namespace D2L.Security.OAuth2 {
+	public static class OAuth2Factory {
+		/// <summary>
+		/// Create an IKeyManagmentService and ITokenSigner given key storage
+		/// and configuration.
+		/// </summary>
+		public static void Create(

```

Comment:
This is a replacement for e.g. `RsaTokenSignerFactory`.

We could have an overload that doesn't take `IPrivateKeyDataProvider` and uses in-memory storage, and then start trimming away some of the old stuff.